src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -138,12 +138,36 @@ def register_options(cls, register):
             default={"python-default": "3rdparty/python/default_lock.txt"},
             help=(
                 "A mapping of logical names to lockfile paths used in your project.\n\n"
-                "For now, things only work properly if you define a single resolve and set "
-                "`[python].experimental_default_resolve` to that value. We are close to "
-                "properly supporting multiple (disjoint) resolves.\n\n"
-                "To generate a lockfile, run `./pants generate-lockfiles --resolve=<name>` or "
-                "`./pants generate-lockfiles` to generate for all resolves (including tool "
-                "lockfiles).\n\n"
+                "Many organizations only need a single resolve for their whole project, which is "
+                "a good default and the simplest thing to do. However, you may need multiple "
+                "resolves, such as if you use two conflicting versions of a requirement in "
+                "your repository.\n\n"
+                "For now, Pants only has first-class support for disjoint resolves, meaning that "
+                "you cannot ergonomically set a `python_source` target, for example, to work "
+                "with multiple resolve. Practically, this means that you cannot yet reuse common "
+                "code, such as util files, across projects using different resolves. Support for "
+                "overlapping resolves is coming soon.\n\n"
+                "If you only need a single resolve, optionally override this option's default if "
+                "you would like to write your project's lockfile to a different path. Make "
+                "sure that `[python].experimental_default_resolve` is set to the same resolve "
+                "name. Then, run `./pants generate-lockfiles --resolve=<name>` to generate the "
+                "lockfile.\n\n"
+                "If you need multiple resolves:\n\n"
+                "  1. Define via this option multiple resolve "
+                "names and their lockfile path. The names should be meaningful to your "
+                "repository, such as `data-science` or `pants-plugins`.\n"
+                "  2. Set the default with "
+                "`[python].experimental_default_resolve`.\n"
+                "  3. Update your `python_requirement` targets with the "
+                "`experimental_compatible_resolves` field to declare which resolve(s) they should "
+                "be available in. (Often you'll set this via the `python_requirements` or "
+                "`poetry_requirements` target generators)\n"
+                "  4. Run `./pants generate-lockfiles --resolve=<name1> --resolve<name2>` to "
+                "generate the lockfiles (or `./pants generate-lockfiles` to generate tool "
+                "lockfiles too). If the results aren't what you'd expect, adjust the prior step.\n"
```

Comment:
```suggestion
                "  4. Run `./pants generate-lockfiles` to regenerate all lockfiles (or "
                "`./pants generate-lockfiles  --resolve=<name1> --resolve<name2>` to generate specific "
                "lockfiles). If the results aren't what you'd expect, adjust the prior step.\n"
```
Explicitly specifying the resolve names is a bit of an advanced case I think. If you're just setting up the lockfiles, you can probably just run `./pants generate-lockfiles`. I expect that it's mostly the `pantsbuild/pants` repo that has ~every tool overridden, and needs the extra precision.